Java: update docs and videos (#5003)
microsoft/vscode-docs: docs/languages/java.md

3. [Community Server Connectors](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-community-server-connector) (for Apache Felix, Karaf, Tomcat, Jetty, etc)
4. [Server Connector](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-server-connector) (for Red Hat Servers, e.g. Wildfly)
5. [Extension Pack for MicroProfile](https://marketplace.visualstudio.com/items?itemName=MicroProfile-Community.vscode-microprofile-pack)
6. [CheckStyle](https://marketplace.visualstudio.com/items?itemName=shengchen.vscode-checkstyle)
7. [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode)

Thanks to the great Java community around VS Code, the list doesn't end there. You can search for more Java extensions easily within VS Code:

1. Go to the **Extensions** view (`kb(workbench.view.extensions)`).
2. Filter the extensions list by typing "java".

This document describes some of the key features included in those Java extensions.

## Getting started

**NOTE:** If you are using VS Code on Windows and want to take advantage of the Windows Subsystem for Linux, see [Developing in WSL](/docs/remote/wsl).

> Before you start, you must have the Java SE Development Kit (JDK) on your local environment. To run the Language Support for Java™ (redhat.java) extension itself, Java SE 11 or above version is required; for projects, the extension supports version 1.5 or above. For how to configure these versions, refer to [Configure JDK](/docs/java/java-project.md#configure-runtime-for-projects).

Reviewer: we will modify this part once we release embedded JDK right?

Author: Yes. 
cc  @testforstephen , we would better create an issue, tracking a todo list after we release platform-specific extensions.

Reviewer 2: Here is the todo list.
https://github.com/microsoft/vscode-java-pack/issues/845

> For developers new to Java or new to VS Code, we provide some tips. Once you've installed the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), you can open the tips from within VS Code with the **Java: Tips for Beginners** command from the Command Palette.

Reviewer: ```suggestion
For developers new to Java or new to VS Code, we do provide some tips in our extensions. Once you've installed the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), you can see the tips using the **Java: Tips for Beginners** command from the Command Palette in VS Code
```
